### PR TITLE
Add initial contributing doc

### PR DESCRIPTION
**Pull request by twpayne**
It came up in yesterday's Cilium meeting that this project does not have a contributing doc. This PR adds an initial one and also tidies up the README a little.

Fixes https://github.com/cilium/ebpf/issues/184

cc @joestringer

### CODE REVIEW

**Comment by tklauser** [comment]
Also see #184 and https://cilium.slack.com/archives/CKC55JL8L/p1607371973036400 for discussion.

**Comment by lmb** [comment]
Could you add a commit that only renames readme.md? Seems like it's throwing off the diff somehow, I can't tell what has changed.

The contribution guidelines sgtm largely, I think we should include that new features have to be accompanied by tests, and that it's best to drop into the slack channel before starting work on a big feature to discuss the design. I'm not super fussed about commit messages, the more detail the better, etc.

**Comment by twpayne** [comment]
Done:
* Added separate rename commit.
* Changes to `README.md` are now in f66f792ff02a3952b46d2ddb414766e045be5d92. It's basically just line wrapping, consistent punctuation, consistent Markdown styling, adding a License section, and adding a `#` before the Slack channel name.
* Updated `CONTRIBUTING.md` to include what @lmb requested.

File: CONTRUBUTING.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+are welcome.
+
+New features must be acompanied by tests. Before starting work on any large
+feature, please [join](https://cilium.herokuapp.com/) the
+[#libbpf-go](https://cilium.slack.com/messages/libbpf-go) channel on Slack to
+discuss the design first.
+
+## Running the tests
+
+Many of the tests require priviliges to set resource limits and load eBPF code.
```
Comment (joestringer) [inline CONTRUBUTING.md:14]:
```suggestion
Many of the tests require privileges to set resource limits and load eBPF code.
```

File: CONTRUBUTING.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# How to contribute
+
+Development is on [GitHub](https://github.com/cilium/ebpf) and contributions in
+the form of pull requests and issues reporting bugs or suggesting new features
+are welcome.
+
+New features must be acompanied by tests. Before starting work on any large
```
Comment (joestringer) [inline CONTRUBUTING.md:7]:
```suggestion
New features must be accompanied by tests. Before starting work on any large
```

**Review by joestringer — CHANGES_REQUESTED** [review]
Couple of minor spelling nits below.

Do we want to mention something about describing the problems as well? eg:

```
When submitting pull requests, consider writing details about what problem you are solving and why the proposed approach solves that problem in commit messages and/or pull request description to help future library users and maintainers to reason about the proposed changes.
```

**Comment by twpayne** [comment]
Thanks for the feedback, PR updated.

**Review by lmb — APPROVED** [review]

**Review by joestringer — APPROVED** [review]

**Comment by twpayne** [comment]
Gentle ping: can this be merged now?

**Comment by thaJeztah** [comment]
This may be specific to people's situation, but leaving it here in case useful to others; if you're running into issues because git doesn't like the case-only filename change (`readme.md` -> `README.md`);

- if you're on a case-insensitive (but case-preserving) filesystem (macOS, Windows): configure git to not ignore case-only changes;

     ```bash
    git config core.ignorecase false
    ```
- if your CI doesn't like this (for example, runc CI failed on the rename https://github.com/opencontainers/runc/pull/2875); adding a `.gitconfig` to your github repository to make the change configuration can help;

    ```config
    [core]
     	ignorecase = false
    ```

**Comment by thaJeztah** [comment]
Hmm.. that config can actually be a bit of a pain as well, because there's no content changes in the readme, git still gets confused when switching branches 😓 

```bash
git checkout master
error: The following untracked working tree files would be overwritten by checkout:
	vendor/github.com/cilium/ebpf/readme.md
Please move or remove them before you switch branches.
Aborting
```

**Pull request merged**